server/routes/bookings/bookingDetailsController.ts

```diff
@@ -1,21 +1,23 @@
 import type { RequestHandler } from 'express'
 import { ValidationChain, matchedData, param, validationResult } from 'express-validator'
-import { VisitService } from '../../services'
+import { BookerService } from '../../services'
 import getPrisonInformation from '../../constants/prisonInformation'
 import paths from '../../constants/paths'
 
 export default class BookingDetailsController {
-  public constructor(private readonly visitService: VisitService) {}
+  public constructor(private readonly bookerService: BookerService) {}
 
   public view(): RequestHandler {
     return async (req, res) => {
       const { booker, bookings } = req.session
 
       const errors = validationResult(req)
-      if (!errors.isEmpty()) {
+      if (!errors.isEmpty() || !bookings?.length) {
         return res.redirect(paths.BOOKINGS.HOME)
       }
 
+      const prisoner = booker.prisoners ? booker.prisoners : await this.bookerService.getPrisoners(booker.reference)
```

Comment:
```suggestion
      const prisoner = booker.prisoners?.[0] ?? (await this.bookerService.getPrisoners(booker.reference))?.[0]
```

(and on line 34, just `prisoner,` instead of `prisoner: prisoner[0],`)

This is because it's still possible for `prisoner` to be empty. Also, keeps naming more appropriate as `prisoner` will be a `Prisoner | undefined` rather than `Prisoner[]`

Ultimately, if there is no prisoner it will go to template as `undefined` but tested and Nunjucks handles this without error (though I guess up to Simon if we display a message rather than just nothing 🤷‍♂️)